network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/revoked/BetaEndedDialog.kt

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2024 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.networkprotection.impl.revoked
+
+import android.app.Activity
+import com.duckduckgo.app.di.AppCoroutineScope
+import com.duckduckgo.common.ui.view.dialog.TextAlertDialogBuilder
+import com.duckduckgo.common.ui.view.dialog.TextAlertDialogBuilder.EventListener
+import com.duckduckgo.common.utils.DispatcherProvider
+import com.duckduckgo.di.scopes.AppScope
+import com.duckduckgo.networkprotection.impl.R.string
+import com.duckduckgo.networkprotection.impl.pixels.NetworkProtectionPixels
+import com.duckduckgo.networkprotection.impl.store.NetworkProtectionRepository
+import com.squareup.anvil.annotations.ContributesBinding
+import javax.inject.Inject
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+
+interface BetaEndedDialog {
+    fun show(activity: Activity)
+}
+
+@ContributesBinding(AppScope::class)
+class RealBetaEndedDialog @Inject constructor(
+    @AppCoroutineScope private val coroutineScope: CoroutineScope,
+    private val dispatcherProvider: DispatcherProvider,
+    private val networkProtectionRepository: NetworkProtectionRepository,
+    private val betaEndStore: BetaEndStore,
```

Comment:
Ditto, I'd remove `BetaEndStore` and use shared prefs directly here as impl detail

Reply:
Done